src/main/java/seedu/address/model/commission/UniqueCommissionList.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    /**
+     * Adds a commission to the list.
+     * The commission must not already exist in the list.
+     */
+    public void add(Commission toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateCommissionException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the commission {@code target} in the list with {@code editedCommission}.
+     * {@code target} must exist in the list.
+     * The commission identity of {@code editedCommission} must not be the same as another existing commission
+     * in the list.
+     */
+    public void setCommission(Commission target, Commission editedCommission) {
+        requireAllNonNull(target, editedCommission);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new CommissionNotFoundException();
+        }
+
+        if (!target.isSameCommission(editedCommission) && contains(editedCommission)) {
+            throw new DuplicateCommissionException();
+        }
+
+        internalList.set(index, editedCommission);
+    }
+
+    /**
+     * Removes the equivalent commission from the list.
+     * The commission must exist in the list.
+     */
+    public void remove(Commission toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new CommissionNotFoundException();
+        }
+    }
+
+    public void setCommissions(UniqueCommissionList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code commissions}.
+     * {@code commissions} must not contain duplicate commissions.
+     */
+    public void setCommissions(List<Commission> commissions) {
+        requireAllNonNull(commissions);
+        if (!commissionsAreUnique(commissions)) {
+            throw new DuplicateCommissionException();
+        }
+
+        internalList.setAll(commissions);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Commission> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Commission> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof UniqueCommissionList // instanceof handles nulls
+                && internalList.equals(((UniqueCommissionList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
+     * Returns true if {@code commissions} contains only unique commissions.
+     */
+    private boolean commissionsAreUnique(List<Commission> commissions) {
```

Comment:
Perhaps this boolean method should be renamed to adhere to the style guide?

From [style guide](https://se-education.org/guides/conventions/java/index.html#naming):

> As much as possible, use a prefix such as is, has, was, etc. for boolean variable/method names so that linters can automatically verify that this style rule is being followed.

For example:
```suggestion
    private boolean areCommissionsUnique(List<Commission> commissions) {
```

Reply:
If you are changing this, we should also change the `customersAreUnique` in `UniqueCustomerList`